### PR TITLE
Implement ConnOpenConfirm datagram

### PR DESCRIPTION
**Pull request by majecty**
~Depends on https://github.com/CodeChain-io/foundry/pull/180~

### CODE REVIEW

File: core/src/ibc/transaction_handler/datagrams.rs [hunk 1]
```diff
@@ -83,6 +85,11 @@ pub enum Datagram {
         proof_height: u64,
         consensus_height: u64,
     },
+    ConnOpenConfirm {
+        identifier: String,
+        proof_ack: Vec<u8>,
```
Comment (junha1) [inline core/src/ibc/transaction_handler/datagrams.rs]: How about `Bytes` for proofs and `ctypes::BlockNumber` for heights? I'll update previous ones (including this).

Comment (majecty) [reply core/src/ibc/transaction_handler/datagrams.rs]: I'll create a new commit that changes Vec<u8> to Bytes in all IBC code.
IMO, we should not use the ctypes::BlockNumber here. The should consider other chain's block number. `ctypes::BlockNumber` is for Foundry only.

Comment (junha1) [reply core/src/ibc/transaction_handler/datagrams.rs]: Ok then I'll leave `Vec<u8>`s. And please contain update to `ctypes::BlockNumber` for *our blocks*. (e.g *making* proof)

Comment (majecty) [reply core/src/ibc/transaction_handler/datagrams.rs]: How about defining a `IBCBlockNumber` type instead?
I'm not sure which method is better, but using a `IBCBlockNumber` type in the all IBC code is easier.

Comment (junha1) [reply core/src/ibc/transaction_handler/datagrams.rs]: Maybe together with `Identifier`

**Pull request merged**